core/common/buffer.cpp

```diff
@@ -0,0 +1,107 @@
+#include <utility>
+
+#include "buffer.hpp"
+#include "hexutil.hpp"
+
+namespace kagome::common {
+
+size_t Buffer::size() const {
+  return _data.size();
+}
+
+Buffer &Buffer::put_uint32(uint32_t n) {
+  _data.reserve(sizeof(n));
```

Comment:
Will this statement actually do something useful? As far as I know, `reserve(..)` increases capacity iff provided argument is greater than the current capacity, so if current capacity is already greater than `sizeof(uint32_t)` it won't do anything